lib/core/maintenance_mode.rb

```diff
@@ -84,9 +84,13 @@ def start_or_stop_maintenance_workload(action)
   def switch_domain_workload(to:)
     step("Switching workload for domain '#{domain_data['name']}' to '#{to}'") do
       cp.set_domain_workload(domain_data, to)
+    end
+
+    step("Waiting for changes to take effect", retry_on_failure: true, wait: 10, max_retry_count: 3) do
```

Comment:
@borela should we be using https://github.com/shakacode/uber_task?

@zzaakiirr is this ready to merge?

Reply:
@justin808 I left a question for @rafaelgomesxyz, I want to get his opinion on approach I took in this PR for checking if domain was updated.

If this is the right way and Rafael approves this PR then we can merge this